show-fixed-kernel-cves.py

```diff
@@ -29,20 +29,23 @@
 from packaging import version
 from optparse import OptionParser
 import urllib.request
+import collections
 
 def print_fixed_linux_cves(from_version_str, to_version_str):
   stream_data_url = "https://raw.githubusercontent.com/nluedtke/linux_kernel_cves/master/data/stream_data.json"
   payload = urllib.request.urlopen(stream_data_url).read()
   streams = json.loads(payload)
   from_version=version.Version(from_version_str)
   to_version=version.Version(to_version_str)
-  cvelist = []
+  cvedict = dict()
   links = []
   for stream, releases in streams.items():
     for release, cves in releases.items():
       if release != "outstanding" and from_version < version.Version(release) <= to_version:
-        cvelist += cves.items()
-  for cve, _ in sorted(cvelist):
+        for k, v in cves.items():
+          cvedict[k] = v
+
+  for cve, _ in collections.OrderedDict(sorted(cvedict.items())).items():
```

Comment:
If we don't need the value of the dict, we maybe can simplify this and the whole patch will be shorter:
```
diff --git a/show-fixed-kernel-cves.py b/show-fixed-kernel-cves.py
index bb5bfb3..0cc5a26 100755
--- a/show-fixed-kernel-cves.py
+++ b/show-fixed-kernel-cves.py
@@ -41,8 +41,8 @@ def print_fixed_linux_cves(from_version_str, to_version_str):
   for stream, releases in streams.items():
     for release, cves in releases.items():
       if release != "outstanding" and from_version < version.Version(release) <= to_version:
-        cvelist += cves.items()
-  for cve, _ in sorted(cvelist):
+        cvelist += cves.keys()
+  for cve in sorted(cvelist):
     links += [f"[{cve}](https://nvd.nist.gov/vuln/detail/{cve})"]
   print(", ".join(links))
 
```

Reply:
How to reproduce:

```
./flatcar-build-scripts/show-fixed-kernel-cves.py -f 5.16.18 -t 5.17.0
```

Out of that range of Kernel versions, it works well.
I also tried figuring out exactly which part of the json file makes the failure, but could not do.

Reply:
Ah, thanks for the example! And of course for working on the fix :)